Sources/BraintreeCore/BTAPIClient.swift

```diff
@@ -147,7 +147,10 @@ import Foundation
         let parameters: [String: Any] = ["configVersion": "3"]
 
         configurationHTTP?.get(configPath, parameters: parameters, shouldCache: true) { [weak self] body, response, error in
-            guard let self else { return }
+            guard let self else {
+                completion(nil, BTAPIClientError.deallocated)
+                return
+            }
 
```

Comment:
We shouldn't need the explicit `self`s below (ex: `self.apiHTTP` and `self.http`)

Reply:
Good call - we were still using self _a lot_ cleaned up here: https://github.com/braintree/braintree_ios/pull/1049/commits/ff0e5256cea5c1a4fac9f5bcfaa12a3468ebc47f